airflow/models/__init__.py

```diff
@@ -3515,6 +3519,7 @@ def get_template_env(self):
 
         env = jinja2.Environment(
             loader=jinja2.FileSystemLoader(searchpath),
+            undefined=self.template_undefined,
```

Comment:
Please correct me if I am wrong, this is a breaking change, isn't it ??

Behavior before this PR is if the variable is not defined it would error.
But after merging this PR: it would not raise an error.

Reply:
The default is Undefined: https://github.com/pallets/jinja/blob/master/jinja2/environment.py#L279

This is still the case if it isn't passed explicitly: https://github.com/apache/airflow/pull/4951/files/65af90c017022c2fb31b24d336e0de45934120cd#diff-664e0133fa172b6e0060f2dd7b2c3504R2990

Reply:
Oh yes, you are right... my bad :) 
Thanks @Fokko

Reply:
No problem, any time @kaxil 👍